src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,125 @@
 package core.basesyntax;
 
-/**
+import java.util.NoSuchElementException;
+
+/*
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
  * реалізації ArrayList (default capacity, newCapacity...)</p>
  */
+
 public class ArrayList<T> implements List<T> {
+    private static final int CAPASITY = 10;
+    private T[] arrayGeneric;
+    private int size;
+
+    public ArrayList() {
+        arrayGeneric = (T[]) new Object[CAPASITY];
+    }
 
     @Override
     public void add(T value) {
-
+        if (size == arrayGeneric.length) {
+            arrayGeneric = resize();
+        }
+        arrayGeneric[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        }
+        if (index > size || index < 0) {
+            throw new ArrayIndexOutOfBoundsException(
+                    "This index is not suitable for the given array");
+        }
 
+        if (size == arrayGeneric.length) {
+            arrayGeneric = resize();
+        }
+        System.arraycopy(arrayGeneric, index, arrayGeneric, index + 1, size - index);
+        arrayGeneric[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        if (size + list.size() >= arrayGeneric.length) {
+            arrayGeneric = resize();
+        }
```

Comment:
do we need this if we are calling `add()` for every element in the list?

Reply:
Sorry, what do you mean add() I have this if in addAll(), P.S - I did everything for the implementation as described in the article: https://habr.com/ru/post/128269/